pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -409,15 +410,14 @@ public synchronized void readEntriesComplete(List<Entry> entries, Object ctx) {
         }
 
         sendMessagesToConsumers(readType, entries);
-
-        readMoreEntries();
     }
 
     protected void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
         int start = 0;
         int entriesToDispatch = entries.size();
         long totalMessagesSent = 0;
         long totalBytesSent = 0;
+        AtomicInteger messagesToConsumer = new AtomicInteger(entries.size());
```

Comment:
why do we need change this file?  method `sendMessagesToConsumers` got override in  `PersistentStickyKeyDispatcherMultipleConsumers` right?

Reply:
This change not to solve the problem of Key_Shared subscription, i initial think this will reduce the reading frequency of ledger in shared subscription.  This will wait for the message to be written to the consumers and then read again.

Reply:
I would suggest remove this change. since we have not meet this related issue in Shared mode, reading frequency of ledger is more controlled by other setting.

This is mainly for message ordering, and ordering is no need for shared mode.
Seems the test errors is related with this change?

Reply:
Agree